lib/bamboo/email.ex

```diff
@@ -67,6 +67,20 @@ defmodule Bamboo.Email do
   ```
   """
 
+  @type address :: String.t | {String.t, String.t}
+
+  @type t :: %__MODULE__{
+    to: nil | address | [address],
+    cc: nil | address | [address],
+    bcc: nil | address | [address],
```

Comment:
I think this should be `nil | any` because the email address can technically be whatever you want as long as it defines the `Bamboo.Formatter` protocol